woocommerce/woocommerce
# [Experimental] Add Product filters overlay settings

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/47039

This PR adds overlay specific settings to the Product Filters block.

### Note

- ~~The overlay preview svg is not added yet. There is just a placeholder. Waiting for design on that.~~ Jarek said we can implement this in the later phase.
- The icon size is using the RangeControl component with some hacky CSS to make it align with the proposed design. This is so we don't need to build a custom component if we don't have to.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Test by developers only

Prereq:
- Ensure you have the WC Beta Tester Plugin installed and go to Tools > WCA Test Helper > Features. Toggle on the experimental-blocks option.

1. Go to the editor and navigate to Product Catalog template.
2. Add the Product Filters (Experimental) block to the editor.
3. Highlight the Product Filters (Experimental) block.
4. You will see some new inspector control options.
5. You should see Overlay with Never, Mobile and Always settings.
6. Selecting Mobile and Always will further expose more settings.
7. You will also see Button settings for displaying as Icon, Icon and label and Icon only.
8. The Icon and label and Icon settings will expose more settings. These two settings will expose an icon style and icon size.
9. Selecting Label only should hide the settings mentioned in step 8.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
This is an experimental block so no changelog is required.

</details>
